File: parallel_projection/parallel.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        w, d, h = input().split()
        w = int(w)
        d = int(d)
        h = int(h)
        a, b, f, g = input().split()
        a = int(a)
        b = int(b)
        f = int(f)
        g = int(g)
        min_length = min(f + a + abs(g - b), g + b + abs(f - a), w - f + w - a + abs(g - b), d - g + d - b + abs(f - a))
        print(h + min_length)
    
    
if __name__ == "__main__":
    main()

File: parallel_projection/parallel.cpp
#include <iostream>
#include <cstdio>
#include <algorithm>
#include <cmath>
#include <vector>

using namespace std;

int main()
{
    int case_num;
    scanf("%d", &case_num);
    for (int i = 0; i < case_num; i++)
    {
        int w, d, h;
        scanf("%d %d %d", &w, &d, &h);
        int a, b, f, g;
        scanf("%d %d %d %d", &a, &b, &f, &g);
        vector<int> possible_lengths = {f + a + abs(g - b), g + b + abs(f - a), w - f + w - a + abs(g - b), d - g + d - b + abs(f - a)};
        int min_length = h + *min_element(possible_lengths.begin(), possible_lengths.end());
        printf("%d\n", min_length);
    }
}
